src/streams/StreamIcon.js

```diff
@@ -8,12 +8,13 @@ type Props = $ReadOnly<{|
   color?: string,
   isPrivate: boolean,
   isMuted: boolean,
+  isSubscribed: boolean,
   size: number,
   style?: TextStyleProp,
 |}>;
 
-export default ({ color, style, isPrivate = false, isMuted = false, size }: Props) => {
-  const StreamIcon = isMuted ? IconMute : isPrivate ? IconPrivate : IconStream;
+export default ({ color, style, isPrivate, isMuted, isSubscribed, size }: Props) => {
+  const StreamIcon = isSubscribed && isMuted ? IconMute : isPrivate ? IconPrivate : IconStream;
```

Comment:
StreamIcon: Don't show as muted when unsubscribed.

This isn't necessary -- before this commit it was already the case that we didn't show a stream as muted when unsubscribed. There was a bug in this in StreamCard, but it was caused by the use of NULL_SUBSCRIPTION and was fixed in the commit that removed that, earlier in this branch.

So this change doesn't seem helpful.

It is kind of awkward that this component's callers are expected to compute `isMuted` as false when there's no subscription. A version of this change that moved that logic inside the component, probably by having it take `stream: Stream, subscription?: Subscription` instead of their individual properties, would improve things.

I'll send a PR with a revert.